agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorMetricExporter.java

```diff
@@ -0,0 +1,91 @@
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.azure.monitor.opentelemetry.exporter;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.MetricDataMapper;
+import com.azure.monitor.opentelemetry.exporter.implementation.logging.OperationLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
+import com.azure.monitor.opentelemetry.exporter.implementation.pipeline.TelemetryItemExporter;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.AggregationTemporalitySelector;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class AzureMonitorMetricExporter implements MetricExporter {
+
+  private static final ClientLogger LOGGER = new ClientLogger(AzureMonitorMetricExporter.class);
+  private static final OperationLogger exportingMetricLogger =
+      new OperationLogger(AzureMonitorMetricExporter.class, "Exporting metric");
+  private final AtomicBoolean stopped = new AtomicBoolean();
+  private final MetricDataMapper mapper;
+  private final TelemetryItemExporter telemetryItemExporter;
+
+  public AzureMonitorMetricExporter(
+      MetricDataMapper mapper, TelemetryItemExporter telemetryItemExporter) {
+    this.mapper = mapper;
+    this.telemetryItemExporter = telemetryItemExporter;
+  }
+
+  @Override
+  public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
+    return AggregationTemporalitySelector.deltaPreferred()
+        .getAggregationTemporality(instrumentType);
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> metrics) {
+    if (stopped.get()) {
+      return CompletableResultCode.ofFailure();
+    }
+
+    List<TelemetryItem> telemetryItems = new ArrayList<>();
+    for (MetricData metricData : metrics) {
+      LOGGER.verbose("exporting metric: {}", metricData);
+      try {
+        mapper.map(metricData, telemetryItems::addAll);
+        exportingMetricLogger.recordSuccess();
+      } catch (Throwable t) {
+        exportingMetricLogger.recordFailure(t.getMessage(), t);
+      }
+    }
+
+    return telemetryItemExporter.send(telemetryItems);
```

Comment:
I need to handle the same way you did on #2298 for trace exporter::export?

Reply:
ya. or alternatively we could not send any telemetry if mapper fails for any of them and returning CompletableResultCode.ofFailure() right away in that case, then we don't need that extra handling that I added in the other PR

Reply:
can you update the AzureMonitorTraceExporter also to match?

Reply:
updated.